distage/distage-typesafe-config/src/main/scala/com/github/pshirshov/izumi/distage/config/typesafe/codec/ConfigReader.scala

```diff
@@ -1,19 +1,24 @@
-package com.github.pshirshov.izumi.distage.config.codec
+package com.github.pshirshov.izumi.distage.config.typesafe.codec
 
 import com.github.pshirshov.izumi.distage.config.model.exceptions.ConfigReadException
+import com.github.pshirshov.izumi.distage.model.reflection.universe.RuntimeDIUniverse._
 import com.typesafe.config.ConfigValueType._
 import com.typesafe.config._
 
 import scala.reflect.ClassTag
 import scala.util.Try
-
 // copypasta from pureconfig.BasicReaders
 
 trait ConfigReader[+T] {
   def apply(configValue: ConfigValue): Try[T]
 }
 
 object ConfigReader {
+
+  def toKeyValue[T : ClassTag : Tag](configReader: ConfigReader[T]) : (SafeType, ConfigReader[T]) = {
```

Comment:
Drop dependency on distage
Accept `scala.reflect.runtime.universe.TypeTag`  instead.

Move `SafeType` to `fundamentals-reflection` as 

```scala
import scala.reflect.api.Universe

case class SafeType[U <: Universe with Singleton](tpe: U#Type)```
```

In `distage-model` `WithDISafeType` leave

```scala
final type SafeType = fundamentals.reflection.SafeType
final val SafeType: fundamentals.reflection.SafeType.type = fundamentals.reflection.SafeType

implicit def ToDeannotate(typ: TypeNative): fundamentals.reflection.SafeType.Deannotate = new Deannotate(typ)
```

Reply:
Done in https://github.com/pshirshov/izumi-r2/pull/372 please rebase